tools/deployment/make_linux_package.sh

```diff
@@ -39,37 +51,53 @@ SYSTEMD_SYSCONFIG_DST_DEB="/etc/default/osqueryd"
 CTL_SRC="$SCRIPT_DIR/osqueryctl"
 PACKS_SRC="$SOURCE_DIR/packs"
 PACKS_DST="/usr/share/osquery/packs/"
-LENSES_LICENSE="${OSQUERY_DEPS}/Cellar/augeas/*/COPYING"
-LENSES_SRC="${OSQUERY_DEPS}/share/augeas/lenses/dist"
+LENSES_LICENSE="${SOURCE_DIR}/libraries/cmake/source/augeas/src/COPYING"
+LENSES_SRC="${SOURCE_DIR}/libraries/cmake/source/augeas/src/lenses"
 LENSES_DST="/usr/share/osquery/lenses/"
 OSQUERY_POSTINSTALL=${OSQUERY_POSTINSTALL:-"$SCRIPT_DIR/linux_postinstall.sh"}
 OSQUERY_PREUNINSTALL=${OSQUERY_PREUNINSTALL:-""}
 OSQUERY_CONFIG_SRC=${OSQUERY_CONFIG_SRC:-""}
 OSQUERY_TLS_CERT_CHAIN_SRC=${OSQUERY_TLS_CERT_CHAIN_SRC:-""}
-OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="${OSQUERY_DEPS}/etc/openssl/cert.pem"
+OSQUERY_TLS_CERT_CHAIN_BUILTIN_SRC="${SCRIPT_DIR}/certs.pem"
 OSQUERY_TLS_CERT_CHAIN_BUILTIN_DST="/usr/share/osquery/certs/certs.pem"
 OSQUERY_EXAMPLE_CONFIG_SRC="$SCRIPT_DIR/osquery.example.conf"
 OSQUERY_EXAMPLE_CONFIG_DST="/usr/share/osquery/osquery.example.conf"
 OSQUERY_LOG_DIR="/var/log/osquery/"
 OSQUERY_VAR_DIR="/var/osquery"
 OSQUERY_ETC_DIR="/etc/osquery"
 
-WORKING_DIR=/tmp/osquery_packaging
-INSTALL_PREFIX=$WORKING_DIR/prefix
-DEBUG_PREFIX=$WORKING_DIR/debug
-
 function usage() {
-  fatal "Usage: $0 -t deb|rpm -i REVISION -d DEPENDENCY_LIST
+  fatal "Usage: $0 -t deb|rpm|pacman|tar
+    [-b|--build] /path/to/build/dir
+    [-d|--dependencies] DEPENDENCY_LIST string
+    [-i|--iteration] REVISION
     [-u|--preuninst] /path/to/pre-uninstall
     [-p|--postinst] /path/to/post-install
     [-c|--config] /path/to/embedded.config
+    [-v|--version] OSQUERY_BUILD_VERSION override
+
   This will generate an Linux package with:
-  (1) An example config /usr/share/osquery/osquery.example.conf
-  (2) An init.d script /etc/init.d/osqueryd
-  (3) A systemd service file /usr/lib/systemd/system/osqueryd.service and
-      a sysconfig file /etc/{default|sysconfig}/osqueryd as appropriate
-  (4) A default TLS certificate bundle (provided by cURL)
-  (5) The osquery toolset /usr/bin/osquery*"
+    (1) An example config /usr/share/osquery/osquery.example.conf
+    (2) An init.d script /etc/init.d/osqueryd
+    (3) A systemd service file /usr/lib/systemd/system/osqueryd.service and
+        a sysconfig file /etc/{default|sysconfig}/osqueryd as appropriate
+    (4) A default TLS certificate bundle (provided by cURL)
+    (5) The osquery toolset /usr/bin/osquery*"
+}
+
+function check_parsed_args() {
+  if [[ -z $PACKAGE_TYPE ]]; then
+    usage
+  fi
+
+  if [[ ! -d $BUILD_DIR ]]; then
+    fatal "Cannot find build dir [-b|--build]: $BUILD_DIR"
+  fi
+
+  FPM_CHECK=$(command -v $FPM || true)
+  if [[ -z $FPM_CHECK ]]; then
```

Comment:
This is an interesting idiom. I'd usually just check the exit code from `command`, or just invoke `fpm` with the `-h` option. I'm curious about the reasoning here.